src/components/tx-flow/useTxStepper.tsx

```diff
@@ -1,17 +1,32 @@
+import { trackEvent } from '@/services/analytics'
 import { useCallback, useState } from 'react'
 
-const useTxStepper = <T extends unknown>(initialData: T) => {
+const useTxStepper = <T extends unknown>(initialData: T, eventCategory?: string) => {
   const [step, setStep] = useState(0)
   const [data, setData] = useState<T>(initialData)
 
-  const nextStep = useCallback((entireData: T) => {
-    setData(entireData)
-    setStep((prevStep) => prevStep + 1)
-  }, [])
+  const nextStep = useCallback(
+    (entireData: T) => {
+      setData(entireData)
+      setStep((prevStep) => {
+        if (eventCategory) {
+          trackEvent({ action: 'Next', category: eventCategory, label: prevStep })
```

Comment:
What's this event?
* `Next`/`Back` is not specific enough. It should be something like `Tx flow next`.
* It should be coming from an enum in `analytics/events`.